software/deploy/deploy_test.go

```diff
@@ -360,6 +366,27 @@ func TestDeployDagsOnlyFailure(t *testing.T) {
 	testUtil.InitTestConfig(testUtil.SoftwarePlatform)
 	deploymentID := "test-deployment-id"
 
+	t.Run("When no deployment id is passed", func(t *testing.T) {
+		featureFlags := &houston.FeatureFlags{
+			DagOnlyDeployment: true,
+		}
+		appConfig := &houston.AppConfig{
+			Flags: *featureFlags,
+		}
+		houstonMock := new(houston_mocks.ClientInterface)
+		houstonMock.On("GetAppConfig", nil).Return(appConfig, nil)
+		dagDeployment := &houston.DagDeploymentConfig{
+			Type: houston.DagOnlyDeploymentType,
+		}
+		deployment := &houston.Deployment{
+			ReleaseName:   "",
+			DagDeployment: *dagDeployment,
+		}
+		houstonMock.On("GetDeployment", mock.Anything).Return(deployment, nil).Once()
+		err := DagsOnlyDeploy(houstonMock, appConfig, "", config.WorkingPath, nil, false)
+		assert.ErrorIs(t, err, errInvalidDeploymentID)
```

Comment:
Add `houstonMock.AssertExpectations(t)` after the assert

Reply:
Done!